### PR TITLE
[#5129] Adjust reports to support cumulative updates feature

### LINKED ISSUE #5129: Feature Request: Cumulative indicator updates support for reports
### What are you trying to do?

Adjust reports to support the cumulative indicator updates feature

### Describe the solution you'd like

Reports that need to be adjusted
- results_indicators_with_map_pdf_reports.py
- results_indicators_excel_report.py
- kickstart_word_report.py
- eutf_org_results_table_excel_report.py
- program_overview_pdf_report.py
- program_period_labels_overview_pdf_report.py
- program_overview_excel_report.py
- project_results_indicators_excel_report.py

### Have you consider alternatives?

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zuhdil**
# TODO / Done

Summarize what has been changed / what has to be done in order to finalize the PR.

- [x] results_indicators_with_map_pdf_reports.py
- [x] results_indicators_excel_report.py
- [x] kickstart_word_report.py
- [x] eutf_org_results_table_excel_report.py
- [x] program_overview_pdf_report.py
- [x] program_period_labels_overview_pdf_report.py
- [x] program_overview_excel_report.py
- [x] project_results_indicators_excel_report.py

# Test plan

What tests are necessary to ensure this works or doesn't break anything working

 - [x] Unit test
 
---

Closes #5129

### CODE REVIEW

File: akvo/rsr/views/py_reports/program_period_labels_overview_pdf_report.py [hunk 1]
```diff
@@ -216,35 +241,54 @@ def progress(self):
...
+        for d in self.period.disaggregations.all():
+            category = d.dimension_value.name.name
+            type = d.dimension_value.value
+            disaggregations[(category, type)] = {
+                'category': category,
+                'type': type,
+                'value': d.value,
+                'numerator': d.numerator,
+                'denominator': d.denominator,
+            }
```
Comment (MichaelAkvo) [inline akvo/rsr/views/py_reports/program_period_labels_overview_pdf_report.py:293]: `DisaggregationData` might be useful here instead of a dict

**Review by MichaelAkvo — CHANGES_REQUESTED** [review]

File: akvo/rsr/tests/views/py_reports/test_cumulative_updates.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+                self.DISAGGREGATION_TYPE_1: {'value': 3},
+                self.DISAGGREGATION_TYPE_2: {'value': 2},
+            }
+        })
+
+
+class ObjectReaderCumulativeUpdateBaseTestCase(CumulativeTestMixin, ABC):
+
+    @abstractmethod
+    def make_project_view(self, project: Project):
```
[the remark this answers is not in the document]
Comment (zuhdil) [reply akvo/rsr/tests/views/py_reports/test_cumulative_updates.py]: Fixed

File: akvo/rsr/views/py_reports/utils.py [hunk 3]
```diff
@@ -499,13 +523,27 @@ def has_qualitative_data(self):
...
         ]
         return {(d.category, d.type): d for d in disaggregations}
 
+    @cached_property
+    def disaggregations(self):
+        disaggregations = {}
+        for d in self._real.disaggregations.all():
+            category = d.dimension_value.name.name
+            type = d.dimension_value.value
+            disaggregations[(category, type)] = {
```
[the remark this answers is not in the document]
Comment (zuhdil) [reply akvo/rsr/views/py_reports/utils.py:538]: These changes will take longer. Better if done in a separate PR.

File: akvo/rsr/views/py_reports/program_period_labels_overview_pdf_report.py (hunk 1 above)
Comment (zuhdil) [reply akvo/rsr/views/py_reports/program_period_labels_overview_pdf_report.py:293]: These changes will take longer. Better if done in a separate PR.

**Pull request merged**